### DISCUSSION TITLE
It suggests only one route, not multiple to choose from

### DISCUSSION THREAD

**Discussion by slrslr** [osmandapp/OsmAnd#12103] [Ideas]
Hello,

going from Android app Waze to OsmAnd 3.9.10 i am confused that OsmAnd does not suggest me multiple routes to choose from. I do not like the default route the app suggest me, i have tried to add mid point , but the app weirdly tell me to return and go the apparently longer road that i do not prefer as more driver focus is needed.

So i need app to suggest multiple routes, show the time and distances so i can compare and select the right one.

**Comment by Lee-Carre** [comment]
You could always plan a route manually, or record a track while taking your preferred route and use that GPX for OsmAnd to follow next time.

Better still, explain why you don't like the suggested route, and what you'd prefer (& why). Perhaps the algorithm can be improved, or OSM map data refined.

Ultimately, a single route being suggested, makes sense. Why would anyone want less than the optimal route? Perhaps the problem is that the suggested route is sub-optimal (based on your local knowledge); in which case let's focus on what info is lacking for the algorithm to recognise it as such.

Failing all that; keep adding way-points along your preferred route, and OsmAnd will follow it.

Else, ensure that route-recalculation is enabled, and navigation origin is your position, start navigation with the desired destination set, and ignore the suggested route (take your preferred route) by deviating from it. If correctly configured, OsmAnd will recognise this, recompute a route from where you are to the destination, and suggest it (with directions, etc.). If that's still not to your liking, keep deviating (following your preferred) until the suggested route matches your preferred.

Suggesting multiple routes seems like a workaround, at best, for not using any of the above features, and to ignore (that is, avoid improvement for) challenging routing problems.

**Reply by TobiPeterG** [reply]
Well, sometimes there are different, similar fast routes.
For example, going to work using the Autobahn is about 3 minutes faster than using a slower road, but the Autobahn is often more stressful to drive.
So I would like to be able to easily select them

Especially should traffic one day become reality, this would be beneficial as I would like to see if there are any traffic jams on one of the routes.

**Comment by hklene** [comment]
1. It's a trade-of between memory consumption and battery consumption for the routing algorithm. The longer (in distance & complexity with lots of turns) a pre-calculated route, the longer (in time & computation) it takes to repair. So often the situation was, that in a foreign city, I missed some twist and got lost on the route, because it kept recalculating, while I was going blindly and once it came up with an alternative, I was already deviating from that alternative again, because it came to late. In the end, I had to stop the car, shut down OSMand and recalculate the route from scratch. This got better with a high-end mobile, but it's still not negligible.
2. Following the instructions makes me feel "dumb". I don't learn about the area, I just focus on getting to the target. But alternative routes activate my brain and let me put the options into perspective. OSMand is kind of explaining to me, why it chose the optimal route and what would happen, if I deviate. Sometimes it is just a minor difference, sometimes it is a complete no-go that requires a U-turn asap.
3. Do I always want the optimal route? BMW has the slogan: "Freude am Fahren" = "Joy of riding". So if I know, I am going to be too early, I can race to the parking lot and take a nap or I accept an appropriate detour: E.g. exit the highway a stop earlier and pass through the town.
4. What is optimal? It's not always about the time or gasoline consumption. It might also be the combination of reaching my target, but at a whim passing by some shop that you only think of in the last minute. Or another app gives me an alarm of cheap gas price and I'd like to take it up.

I think, there are many more reasons, why deviating from the route happens all the time - unmapped road blocks / construction sites ... Route repair and reorientation takes time and sometimes requires the driver giving hints - I have to tell OSMand not to make me U-turn and attempt to go over that blocked bridge again, that I just turned away from. Fiddling with intermediate targets is doable for the passenger, but not for the driver unless you stop first.

My guess would be, that the routing algorithm on finding the optimal route would evaluate alternatives anyway - just to prove, they are not optimal. So even, if it didn't work out all the details on the alternative, it already knows, it exists and has some higher cost. With this feature request, OSMand could as a first step expose, what it already knows, so the driver can make a better decision without stopping and assessing the wealth of options.

The key is to keep it simple and only offer alternatives, that are "next to optimal", but might offer different benefits: Another gas station, a scenic route, a toilet stop, less traffic lights, ... those criteria are hard to determine and also Google doesn't yet try to guess, why people like some specific detour more than the "optimal" route. But having alternative suggestions in place would be a building block, for what an AI might guess in the future.

**Reply by sjvudp** [reply]
TomTom and Google Maps both offer alternate routes initially; it's also important for real-time routing when an alternate route is faster due to traffic or accidents.

See for example this route: https://plan.tomtom.com/de/route/plan?p=48.95427,12.36915,10.36z&q=Straubing&r=(costModel:FASTEST,routingProvider:GLOBAL,sorted:(h~V48.880064~J12.570267~Vaddr~JStraubing,h~V49.03181~J12.22214~Vaddr~JWalhallastra%2703V%27e_48~E_93093_Donaustauf),travelMode:CAR,vehicleParameters:(axleWeight:-+,height:-+,length:-+,maxSpeed:-+,vehicleModelId:-+,weight:-+,width:-+))

The main route is dark blue and two alternatives are gray.

![grafik](https://github.com/user-attachments/assets/82312612-2cb2-4bc3-904a-6830c7fbba95)

**Comment by indykoning** [comment]
I'm missing the similar idea myself as well switching from Google Maps. But in a slightly different way. 

I have a road which is technically faster, but due to the sheer amount of roundabouts if there's a large truck ahead of me it will be slower and less enjoyable to drive. 

What I had in Maps is that I could see an alternative route is 1 or 2 minutes slower with current traffic. Then I can decide to take that route because the current route will in reality be slower than that alternate route.

**Reply by sjvudp** [reply]
I'm afraid the app cannot detect the truck in front of you, but it could consider the roundabouts:
My old TomTom had a simulation mode where the car virtually drives the route: There you could clearly see the car decelerate in front of each roundabout, even if there was no speed limit there. So it would "kind of consider" those...

**Comment by 0-5-0** [comment]
Is adding multiple route suggestions in the plans?

Having to manually add waypoints just to try to "force" the route suggestion to use my preferred route is one of the things I disliked most when using OSMAnd, then making sure it didn't add the waypoint on the wrong side of the road because that caused other problems of its own.

Other navigation apps as mentioned offer a better routing experience by offering 3 routes but it would be amazing to having this in OSMAnd. If performance is a concern then it could be made into an optional selection (offer 1 route or offer 3 routes via settings).

**Reply by DmitryAlexei** [reply]
Issue 
- https://github.com/osmandapp/OsmAnd/issues/1294

**Comment by sjvudp** [comment]
Well I discovered another variant of the problem when trying bicycle navigation: I started on a main road when I was close to the destination. Intending to use the "optimal" (in terms of time) route I followed the navigation instructions that first lead me off the main road into some odd pattern of minor roads. Eventually I had missed the right road (also due to the lack of a phone holder on the rent bike). Later I discovered that I had set the option to avoid main roads for the bike.
However if the suggested route is significantly longer than the optimal route, shouldn't OSMAND mention that fact and maybe ask whether to use the shorter route instead?

**Comment by 002gRTEah3mRYH** [comment]
I wish we can have that feature
